src/simulator/tests/halo_simulator_test.py

```diff
@@ -90,6 +90,93 @@ def test_simulated_reach_by_spend_with_privacy(self, mock_laplace_mechanism):
         )
         self.assertTrue(reach_point.reach(1) >= 0)
 
+    @parameterized.named_parameters(
+        # testcase_name, num_publishers, spends, regions, expected
+        {
+            "testcase_name": "with_empty_regions",
+            "num_publishers": 2,
+            "spends": [0.005, 0.01],
+            "regions": {},
+            "expected": [
+                ReachPoint([0, 0], [0], [0.005, 0]),
+                ReachPoint([0, 0], [0], [0, 0.005]),
+                ReachPoint([0, 0], [0], [0.005, 0.005]),
+            ],
+        },
+        {
+            "testcase_name": "without_active_publishers",
+            "num_publishers": 2,
+            "spends": [0, 0],
+            "regions": {},
+            "expected": [],
+        },
+        {
+            "testcase_name": "with_1_active_pub_from_2_pubs",
+            "num_publishers": 2,
+            "spends": [0.04, 0.02],
+            "regions": {1: [2]},
+            "expected": [
+                ReachPoint([3, 0], [2], [0.04, 0]),
+                ReachPoint([0, 0], [0], [0, 0.04]),
```

Comment:
Shouldn't the spend vector in this case be [0, 0.02]?

Reply:
Yes it should. Thanks for pointing out.

Reply:
Fixed.